src/index.js

```diff
@@ -1,72 +1,208 @@
+
+    // valid dates should not be converted
+    if (dateTimeRegex.test(value)) {
+        return value;
+    }
 
-  const {
+    return convertDateToString(new Date(value));
+};
+
+/**
+ * Converts a datetime string without timezone to a date object
+ * with timezone, using the browser timezone.
+ *
+ * @param {String} value Date string, formatted as yyyy-MM-ddThh:mm
+ * @return {Date}
+ */
+const dateTimeParser = (value) => value ? new Date(value) : '';
+
+const sanitizeRestProps = ({
+    allowEmpty,
+    alwaysOn,
+    basePath,
+    component,
+    defaultValue,
+    format,
+    formClassName,
+    initialValue,
+    initializeForm,
+    input,
+    isRequired,
+    label,
+    limitChoicesToValue,
+    locale,
+    meta,
     options,
+    optionText,
+    optionValue,
+    parse,
+    record,
+    resource,
+    source,
+    textAlign,
+    translate,
+    translateChoice,
+    labelTime,
+    ...rest
+}) => rest;
+
+const Picker = ({
+    PickerComponent,
+    format = dateTimeFormatter,
     label,
+    options,
     source,
     resource,
-    className,
+    helperText,
+    margin = 'dense',
+    onBlur,
+    onChange,
+    onFocus,
+    parse = dateTimeParser,
+    validate,
+    variant = 'filled',
+    defaultValue,
+    providerOptions: {utils, locale},
+    pickerVariant = 'dialog',
+    ...rest
+}) => {
+    const translate = useTranslate();
+    const {
+        id,
+        input,
     isRequired,
-    providerOptions,
-  } = fieldProps;
+        meta: {error, touched},
+    } = useInput({
+        format,
+        onBlur,
+        onChange,
+        onFocus,
+        parse,
+        resource,
+        source,
+        validate,
+        /* type: 'datetime-local', */
+        ...rest,
+    });
 
-  const { input, meta } = useInput({ source });
-  
-  const { touched, error } = meta;
-  
   const handleChange = useCallback(value => {
     Date.parse(value) ? input.onChange(value.toISOString()) : input.onChange(null);
   }, []);
 
   return (
-    <div className="picker">
-      <MuiPickersUtilsProvider {...providerOptions}>
+      <MuiPickersUtilsProvider utils={utils || DateFnsUtils} locale={locale}>
         <PickerComponent
-          {...options}
+            id={id}
           label={<FieldTitle
             label={label}
             source={source}
             resource={resource}
             isRequired={isRequired}
           />}
-          margin="normal"
+            InputLabelProps={{
+                shrink: true,
+            }}
+            variant={pickerVariant}
+            inputVariant={variant}
+            margin={margin}
           error={!!(touched && error)}
-          helperText={touched && error}
-          className={className}
+            helperText={
+                <InputHelperText
+                  touched={touched}
+                  error={error}
+                  helperText={helperText}
+                />
+            }
+            clearLabel={translate('ra.action.clear_input_value')}
```

Comment:
clearLabel and cancelLabel props make some warning on browser console. could you change the name of these props?

```
react-dom.development.js:88 Warning: React does not recognize the `cancelLabel` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `cancellabel` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```